### ISSUE TITLE
[Bug]: `add_electrodes` throws error when writing two recordings with differing additional electrode properties 

### ISSUE THREAD

**Issue by felixp8** [catalystneuro/neuroconv#556]
### What happened?

Unsure to what extent this behavior is intended, so it may not be a bug, but I'm hitting an error from the check in [these lines](https://github.com/catalystneuro/neuroconv/blob/main/src/neuroconv/tools/spikeinterface/spikeinterface.py#L256-L258) in `add_electrodes` when trying to write two separate recordings.

Issue seems to be:
1. Both recording interfaces separately modify `metadata["Ecephys"]["Electrodes"]` to add (non-overlapping) property descriptions
2. `NWBConverter.get_metadata` combines these with `deep_dict_update` into a single list
3. `add_electrodes` uses this merged list to create `property_descriptions` dict
4. Neither recording has all of the properties now in `property_descriptions` so the check in the lines linked above fails

I can get around this by manually adding any missing properties to the recordings but I'm hoping there's a better solution.

### Steps to Reproduce

```python
import numpy as np
from neuroconv.tools.testing.mock_interfaces import MockRecordingInterface
from neuroconv import NWBConverter

class DummyRecordingInterface1(MockRecordingInterface):
    def __init__(self, verbose: bool = True, es_key: str = "ElectricalSeries", **kwargs):
        super().__init__(verbose=verbose, es_key=es_key, **kwargs)
        self.recording_extractor.set_property(
            key="property_a",
            ids=self.recording_extractor.get_channel_ids(),
            values=np.full(self.recording_extractor.get_num_channels(), np.nan),
        )

    def get_metadata(self):
        metadata = super().get_metadata()
        metadata["Ecephys"]["Electrodes"] = [
            dict(name="property_a", description="no description.")
        ]
        return metadata

class DummyRecordingInterface2(MockRecordingInterface):
    def __init__(self, verbose: bool = True, es_key: str = "ElectricalSeries", **kwargs):
        super().__init__(verbose=verbose, es_key=es_key, **kwargs)
        self.recording_extractor.set_property(
            key="property_b",
            ids=self.recording_extractor.get_channel_ids(),
            values=np.full(self.recording_extractor.get_num_channels(), np.nan),
        )

    def get_metadata(self):
        metadata = super().get_metadata()
        metadata["Ecephys"]["Electrodes"] = [
            dict(name="property_b", description="no description.")
        ]
        return metadata

class DummyNWBConverter(NWBConverter):
    data_interface_classes = dict(
        Recording1=DummyRecordingInterface1,
        Recording2=DummyRecordingInterface2,
    )

source_data = dict(
    Recording1=dict(es_key="ElectricalSeries1"),
    Recording2=dict(es_key="ElectricalSeries2"),
)

converter = DummyNWBConverter(source_data=source_data)
converter.run_conversion()
```

### Traceback

```python
Traceback (most recent call last):
  File "/home/jovyan/scripts/test_add_electrodes.py", line 49, in <module>
    converter.run_conversion()
  File "/home/jovyan/neuroconv/src/neuroconv/nwbconverter.py", line 160, in run_conversion
    self.add_to_nwbfile(nwbfile_out, metadata, conversion_options)
  File "/home/jovyan/neuroconv/src/neuroconv/nwbconverter.py", line 114, in add_to_nwbfile
    data_interface.add_to_nwbfile(
  File "/home/jovyan/neuroconv/src/neuroconv/datainterfaces/ecephys/baserecordingextractorinterface.py", line 335, in add_to_nwbfile
    add_recording(
  File "/home/jovyan/neuroconv/src/neuroconv/tools/spikeinterface/spikeinterface.py", line 752, in add_recording
    add_electrodes_info(recording=recording, nwbfile=nwbfile, metadata=metadata)
  File "/home/jovyan/neuroconv/src/neuroconv/tools/spikeinterface/spikeinterface.py", line 725, in add_electrodes_info
    add_electrodes(recording=recording, nwbfile=nwbfile, metadata=metadata)
  File "/home/jovyan/neuroconv/src/neuroconv/tools/spikeinterface/spikeinterface.py", line 258, in add_electrodes
    raise ValueError(f"{extra_descriptions} are not available in the recording extractor, set them first")
ValueError: ['property_b'] are not available in the recording extractor, set them first
```

### Operating System

Linux

### Python Executable

Python

### Python Version

3.9

### Package Versions

[issue_environment.txt](https://github.com/catalystneuro/neuroconv/files/12500077/issue_environment.txt)

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/catalystneuro/neuroconv/blob/main/.github/CODE_OF_CONDUCT.rst)
- [X] Have you ensured this bug was not already [reported](https://github.com/catalystneuro/neuroconv/issues)?

**Comment by CodyCBakerPhD** [comment]
@h-mayorquin The inner workings of this ought to be right up your alley with regards to the append feature of the electrodes table for multiple SpikeInterface recording objects - any idea how to fix (and add tests) to catch this going forward?

@felixp8 Thanks for your patience; I suppose just stick to populating the missing properties manually for the time being, but I'm fairly certain our original design for the `add_electrodes` function is that it would work in situations just like this

**Comment by h-mayorquin** [comment]
Thanks for such a clean example when reporting the bug, @felixp8.

I think that those lines are outdated and after some thinking I can't see what will be the ill of removing them. They were introduced a while ago and I just carried them out after the refactoring that I did:

https://github.com/catalystneuro/neuroconv/commit/402123086a05600c881298fe12ace8b685b459d9

I think we should just remove those lines:
https://github.com/catalystneuro/neuroconv/blob/main/src/neuroconv/tools/spikeinterface/spikeinterface.py#L256-L258

Properties in the general electrode metadata might exists that are not in that specific recoder as your use case illustrates. As long as the `add_electrodes` function is properly appending them right to each other if they share channel_names there should not be a problem.
